src/file-upload/internal.tsx

```diff
@@ -105,97 +113,98 @@ function InternalFileUpload(
   const invalid = restProps.invalid || formFieldContext.invalid || hasError;
 
   return (
-    <InternalSpaceBetween
-      {...baseProps}
-      size="xs"
-      className={clsx(baseProps.className, styles.root)}
-      __internalRootRef={__internalRootRef}
-    >
-      <InternalBox>
-        {isDropzoneVisible ? (
-          <Dropzone onChange={handleFilesChange}>{i18nStrings.dropzoneText(multiple)}</Dropzone>
-        ) : (
-          <FileInput
-            ref={ref}
-            accept={accept}
-            ariaRequired={ariaRequired}
-            multiple={multiple}
-            onChange={handleFilesChange}
-            value={value}
-            {...restProps}
-            ariaDescribedby={ariaDescribedBy}
-            invalid={invalid}
-          >
-            {i18nStrings.uploadButtonText(multiple)}
-          </FileInput>
-        )}
-
-        {(constraintText || errorText || warningText) && (
-          <div className={styles.hints}>
-            {errorText && (
-              <FormFieldError id={errorId} errorIconAriaLabel={i18nStrings?.errorIconAriaLabel}>
-                {errorText}
-              </FormFieldError>
-            )}
-            {showWarning && (
-              <FormFieldWarning id={warningId} warningIconAriaLabel={i18nStrings?.warningIconAriaLabel}>
-                {warningText}
-              </FormFieldWarning>
-            )}
-            {constraintText && (
-              <ConstraintText id={constraintTextId} hasValidationText={!!errorText || !!warningText}>
-                {constraintText}
-              </ConstraintText>
-            )}
-          </div>
-        )}
-      </InternalBox>
-
-      {!multiple && value.length > 0 ? (
+    <div style={{ display: 'contents' }} ref={tokenListRef}>
```

Comment:
Introduced a new element to that can pass the token list ref. The ref is used to query the list items, show-more button, and outside elements to dispatch the focus to.